src/main/webapp/meal-page-script.js

```diff
@@ -13,7 +13,13 @@
 // limitations under the License.
 
 function fetchMealInfo() {
-    fetch('/meal').then(response => response.json()).then((meal) => 
+    // use mapping /meal.html?id=<id> 
+    // fetches form server by action meal/<id>
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = urlParams.get("id");
```

Comment:
Probably a good idea to check `null` case. Make a request to default value for example.